tsconfig.json

```diff
@@ -95,5 +95,5 @@
     // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
     "skipLibCheck": true /* Skip type checking all .d.ts files. */
   },
-  "exclude": ["node_modules"]
+  "exclude": ["node_modules", "demos"]
```

Comment:
Should we be re-enabling this in the future, or is the plan to keep it permanently excluded?

Reply:
They're fixture sites generated by Remix, so it doesn't seem like we should be testing/linting/typechecking them.

Also, to be clear, we weren't typechecking `demos/` before either. It's a bit confusing. So, this is a monorepo where `packages/*` and `demos` are all monorepo packages. Each of the packages within `packages/*` was individually typechecking its own source (confusingly, implicitly doing this via the `--dts` passed to this `tsup` thing I'd never heard of before) but there was no typechecking in the workspace root, so `demos/*` wasn't typechecked but also test files and .ts config files weren't typechecked. In this PR I've added typechecking at the root but excluding `demos/`.

Reply:
Also, `demos/` should maybe be renamed `fixtures/`...